jupyter_server/services/contents/filemanager.py

```diff
@@ -55,6 +56,31 @@ def _validate_root_dir(self, proposal):
             raise TraitError("%r is not a directory" % value)
         return value
 
+    @default("preferred_dir")
+    def _default_preferred_dir(self):
+        try:
+            value = self.parent.preferred_dir
+        except AttributeError:
+            pass
+        else:
+            if value is not None:
+                warnings.warn(
+                    "ServerApp.preferred_dir config is deprecated in jupyter-server 2.0. Use ContentsManager.preferred_dir with a relative path instead",
+                    DeprecationWarning,
```

Comment:
A quick search through the code-base shows that `DeprecationWarning` is the general pattern that we use for deprecations. I think we should be consistent in how we flag deprecations, so the question is whether this deprecation is different in any way to warrant a different warning type.

Reply:
I agree with being consistent.  I just don't see the deprecation warning when I run until changing the category, and, per the docs, `DeprecationWarning` is _ignored by default, unless triggered by code in \_\_main\_\__ - so I believe I'm seeing correct behavior.  Perhaps this is a Windows thing or something with how modules are loaded?  I ran into the [same thing when we needed to deprecate `KernelManager.cleanup()`](https://github.com/jupyter/jupyter_client/blob/6b1aa1cdd39daea559966da75f265ed7a8998602/jupyter_client/manager.py#L415-L416) due to this same issue.  This warnings stuff seems to have lots of knobs and perhaps my system is missing what it takes to surface `DeprecationWarning`.

I think it's more important for _users_ to see this than _devs_ so however it can be surfaced is fine - it just needs to be surfaced consistently.

Reply:
You can surface deprecation warnings by setting the env var `PYTHONWARNINGS=default`. That said, I do see your point about this deprecation being targeted at users, and not devs, which is consistent with using `FutureWarning` (https://docs.python.org/3/library/warnings.html#warning-categories), so I'll accept this suggestion. We could probably make a  separate pass to see if there are other deprecations that should be changed as well.